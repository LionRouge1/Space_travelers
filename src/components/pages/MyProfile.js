/* eslint-disable jsx-quotes */
import React from 'react';
import JoinMission from '../mission/JoinMission';
import ReservedRockets from '../rocket/reservedRockets';
import '../../style/profile.css';

const MyProfile = () => (
  <div className='profile'>
    <ReservedRockets />
    <JoinMission />
  </div>
);

export default MyProfile;
