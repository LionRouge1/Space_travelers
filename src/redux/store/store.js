/* eslint-disable no-underscore-dangle */
import {
    combineReducers,
    createStore,
    applyMiddleware,
    compose,
} from 'redux';
import thunk from 'redux-thunk';
import RocketReducer from '../reducers/rocket';

const rootReducer = combineReducers({
    rockets: RocketReducer,
});

const store = createStore(
    rootReducer,
    compose(
        applyMiddleware(thunk),
    ),
);

export default store;