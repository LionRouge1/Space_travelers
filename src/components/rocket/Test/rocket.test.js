import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../../redux/ConfigStore';
import RocketList from '../rocketList';
import MyProfile from '../../pages/MyProfile';

describe('Main page snapshot tests', () => {
  it('RocketList Page renders correctly', () => {
    const rocketsPage = renderer
      .create(
        <Provider store={store}>
          <RocketList />
        </Provider>,
      )
      .toJSON();
    expect(rocketsPage).toMatchSnapshot();
  });

  it('Missions Page renders correctly', () => {
    const missionsPage = renderer
      .create(
        <Provider store={store}>
          <missions />
        </Provider>,
      )
      .toJSON();
    expect(missionsPage).toMatchSnapshot();
  });

  it('MyProfile Page renders correctly', () => {
    const MyProfilePage = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();
    expect(MyProfilePage).toMatchSnapshot();
  });
});
