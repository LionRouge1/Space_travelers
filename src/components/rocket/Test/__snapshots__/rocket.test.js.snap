// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page snapshot tests Missions Page renders correctly 1`] = `<missions />`;

exports[`Main page snapshot tests MyProfile Page renders correctly 1`] = `
<main
  className="profile"
>
  <div
    className="reservedRocketContainer"
  >
    <h2
      className="heading"
    >
      Reserved Rockets
    </h2>
    <ul
      className="reservedRockets"
    />
  </div>
  <div
    className="joined"
  >
    <h2
      className="heading"
    >
      Joined Missions
    </h2>
    <ul />
  </div>
</main>
`;

exports[`Main page snapshot tests RocketList Page renders correctly 1`] = `
<div
  className="rocketListContainer"
/>
`;
